scripts/simplify_network.py

```diff
@@ -743,6 +746,141 @@ def drop_isolated_nodes(n, threshold):
     return n
 
 
+def transform_to_gdf(n, network_crs):
+    buses_df = n.buses.copy()
+    buses_df["bus_id"] = buses_df.index
+
+    # load data are crucial to deal with sub-networks
+    buses_df = buses_df.join(n.loads_t.p_set.sum().T.rename("load"))
+    buses_df["load_in_subnetw"] = buses_df.groupby(["country", "sub_network"])[
+        "load"
+    ].transform("sum")
+    buses_df["load_in_country"] = buses_df.groupby(["country"])["load"].transform("sum")
+    buses_df["sbntw_share_of_country_load"] = buses_df.apply(
+        lambda row: (
+            row.load_in_subnetw / row.load_in_country if row.load_in_country > 0 else 0
+        ),
+        axis=1,
+    )
+    buses_df["is_backbone_sbntw"] = (
+        buses_df.groupby(["country"], as_index=True)[
+            "sbntw_share_of_country_load"
+        ].transform("max")
+        <= buses_df["sbntw_share_of_country_load"]
+    )
+
+    gdf_buses = gpd.GeoDataFrame(
+        buses_df,
+        geometry=gpd.points_from_xy(buses_df.x, buses_df.y),
+        crs=network_crs,
+    )
+    return gdf_buses
+
+
+def merge_into_network(n, threshold, aggregation_strategies=dict()):
+    """
+    Find isolated AC nodes and sub-networks in the network and merge those of
+    them which have load value and a number of buses below than the specified
+    thresholds into a backbone network.
+
+    Parameters
+    ----------
+    n : PyPSA.Network
+        Original network
+    threshold : float
+        Load power used as a threshold to merge isolated nodes
+    aggregation_strategies: dictionary
+        Functions to be applied to calculate parameters of the aggregated grid
+
+    Returns
+    -------
+    modified network
+    """
+    # keep original values of the overall load and generation in the network
+    # to track changes due to drop of buses
+    generators_mean_origin = n.generators.p_nom.mean()
+    load_mean_origin = n.loads_t.p_set.mean().mean()
+
+    network_crs = snakemake.params.geo_crs
+
+    n.determine_network_topology()
+
+    n_buses_gdf = transform_to_gdf(n, network_crs=network_crs)
+
+    # do not merge sub-networks spanned through a number of countries
+    n_buses_gdf["is_multicnt_subntw"] = n_buses_gdf.sub_network.map(
+        n_buses_gdf.groupby(["sub_network"])["country"].apply(
+            lambda x: len(pd.unique(x)) > 1
+        )
+    )
```

Comment:
Not sure if you need the double map; probably the following should work or something like that
```suggestion
    n_buses_gdf["is_multicnt_subntw"] = n_buses_gdf.sub_network.map(
         n_buses_gdf.groupby(["sub_network"]).country.unique().count() > 1
     )
```
May that work?

However, I'm wondering if we need to fill that value at the gdf and we can work with a df for subnetworks, like:
```
df_subn = n_buses_gdf.groupby(["subnetwork"]).apply(lambda x:
   {
       "is_multicnt_subntw": x.country.unique() > 1,
       "total_demand": x.load.sum(),
   }
)
```

However, I see that in the buses gdf you filled a column with the demand of the subnetwork, so that may not be needed actually

Reply:
Agree on the revision on `map().apply()`, thanks! :D

Not sure I get your idea with the sub-networks dataframe. I thought, we aim to keep all the data in a single dataframe in the spirit of [the tidy data](https://r4ds.had.co.nz/tidy-data.html) concept, which actually encourages a certain degree of duplication. It gives a lot of power and flexibility, while allowing to stay in the same "variables space". If we add an additional dataframe, it means a need to switch attention between two objects, and each of them has an own set of variables. To me, it sounds a bit more complicated.

I'm also not sure that having two objects would be more effective in terms of performance, as compared with having an additional column. Have I missed anything? :)